### PR TITLE
GSoC: Add Proposal for KubeBuilder - Helper to upgrade projects

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description

We want to add this proposal.

### CODE REVIEW

File: summerofcode/2023.md [hunk 1]
```diff
@@ -26,6 +26,21 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome:
+  - Develop a helper to achieve the goal that makes it easier for users to upgrade their projects.
+  - Interfact with the community: make sure the roadmap is generally expected, continuously receive feedbacks to move forward the project.
+  - Make the helpers properly covered with tests and documentation
+  - Demostrate the feature into the community meeting
+- Recommended Skills: Go
+- Mentor(s): [Camila Macedo](https://github.com/camilamacedo86), [Varsha](https://github.com/varshaprasad96), [Tony (TianYi)](https://github.com/Kavinjsir)
+- Expected project size: 350h
+- Difficulty: Medium
+- Design Proposal doc (URL): https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/helper_to_upgrade_projects_by_rescaffolding.md
```
Comment (aliok) [inline summerofcode/2023.md:42]: This document is very good but it is still better to have a Github issue for the project idea for allowing discussions.
Can you please create an issue and link it here?

See the template https://github.com/cncf/mentoring/pull/844/files#diff-c293d7a32a3366987d0fc8032fc59789df1f5b33c8a6cc6de29872b0d4e05204R22

File: summerofcode/2023.md [hunk 2]
```diff
@@ -26,6 +26,21 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome:
+  - Develop a helper to achieve the goal that makes it easier for users to upgrade their projects.
+  - Interfact with the community: make sure the roadmap is generally expected, continuously receive feedbacks to move forward the project.
```
Comment (aliok) [inline summerofcode/2023.md:35]:
```suggestion
  - Interact with the community: make sure the roadmap is generally expected, continuously receive feedbacks to move forward the project.
```

File: summerofcode/2023.md [hunk 3]
```diff
@@ -26,6 +26,21 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+  - Make the helpers properly covered with tests and documentation
+  - Demostrate the feature into the community meeting
+- Recommended Skills: Go
+- Mentor(s): [Camila Macedo](https://github.com/camilamacedo86), [Varsha](https://github.com/varshaprasad96), [Tony (TianYi)](https://github.com/Kavinjsir)
```
Comment (aliok) [inline summerofcode/2023.md:39]: Could you add email addresses of the mentors? You can have a look at other ideas for formatting.

**Review by aliok — COMMENTED** [review]
@camilamacedo86 Very nice to see your name again :) I hope you're doing well!

Added a few comments

File: summerofcode/2023.md (hunk 1 above)
Comment (camilamacedo86) [inline summerofcode/2023.md:42]:
```suggestion
- Design Proposal doc (URL): https://github.com/kubernetes-sigs/kubebuilder/issues/3244
```

File: summerofcode/2023.md [hunk 4]
```diff
@@ -26,6 +26,21 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome:
+  - Develop a helper to achieve the goal that makes it easier for users to upgrade their projects.
+  - Interact with the community: make sure the roadmap is generally expected, continuously receive feedbacks to move forward the project.
+  - Make the helpers properly covered with tests and documentation
+  - Demostrate the feature into the community meeting
```
Comment (camilamacedo86) [inline summerofcode/2023.md:37]:
```suggestion
  - Develop a helper to achieve the goal, making it easier for users to upgrade their projects.
  - Interact with the community: make sure the roadmap is generally expected, and continuously receive feedback to move forward with the project.
  - Make the helpers covered adequately with tests and documentation
  - Demonstrate the feature in the community meeting
```

File: summerofcode/2023.md [hunk 5]
```diff
@@ -26,6 +26,21 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome:
+  - Develop a helper to achieve the goal, making it easier for users to upgrade their projects.
+  - Interact with the community: make sure the roadmap is generally expected, and continuously receive feedback to move forward with the project.
+  - Make the helpers covered adequately with tests and documentation
+  - Demonstrate the feature in the community meeting
+- Recommended Skills: Go
+- Mentor(s): [Camila Macedo](https://github.com/camilamacedo86), [Varsha](https://github.com/varshaprasad96), [Tony (TianYi)](https://github.com/Kavinjsir)
```
Comment (camilamacedo86) [inline summerofcode/2023.md:39]:
```suggestion
- Mentor(s): [Camila Macedo](https://github.com/camilamacedo86) (camilamacedo86 AT gmail DOT com), [Varsha](https://github.com/varshaprasad96) (vnarsing AT redhat DOT com), [Tony (TianYi)](https://github.com/Kavinjsir) (kavinjsir AT gmail DOT com)
```


Mentor(s): Pierangelo Di Pilato @pierDipi (pierdipi AT redhat DOT com), Ali Ok @aliok (aliok AT redhat DOT com)

**Comment by camilamacedo86** [comment]
Hi @aliok,

> @camilamacedo86 Very nice to see your name again :) I hope you're doing well!

Thank you for your amazing review and time. All suggestions were addressed. Please feel free to check it out again.
I also hope that you are doing well. It is amazing to be able to see you again :-)

**Comment by aliok** [comment]
LGTM

@nate-double-u  I don't see anything specific for Kubebuilder in [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv) file but I see mentors' names in Kubebuilder's [OWNERS_ALIASES file](https://github.com/kubernetes-sigs/kubebuilder/blob/master/OWNERS_ALIASES). 
So, 👍 from me if that counts :)

**Comment by camilamacedo86** [comment]
Hi @aliok,

> @nate-double-u I don't see anything specific for Kubebuilder in [project-maintainers.csv](https://github.com/cncf/foundation/blob/main/project-maintainers.csv) file but I see mentors' names in Kubebuilder's [OWNERS_ALIASES file](https://github.com/kubernetes-sigs/kubebuilder/blob/master/OWNERS_ALIASES).
So, 👍 from me if that counts :)

[Varsha](https://github.com/varshaprasad96) (vnarsing AT redhat DOT com) and I ( [Camila Macedo](https://github.com/camilamacedo86) (camilamacedo86 AT gmail DOT com), ) are KubeBuilder admins

[Tony (TianYi)](https://github.com/Kavinjsir) (kavinjsir AT gmail DOT com) will par with us this year. He was a student who did a project in this program last year and has contributed to the project. He also should have his name in the OWNER files. He has achieved the status of Reviewer and probably Approver already.

**Comment by aliok** [comment]
> [Tony (TianYi)](https://github.com/Kavinjsir) (kavinjsir AT gmail DOT com) will par with us this year. He was a student who did a project in this program last year and has contributed to the project. He also should have his name in the OWNER files. He has achieved the status of Reviewer and probably Approver already.

That's no problem. CNCF needs 1 responsible maintainer for oversight anyway.

BTW, it is great to see a student of last year becoming a mentor this year. Awesome job @Kavinjsir 😍

**Pull request merged**